File: aa-integration-tests/tests/fixtures/agents/typescript/_shared.ts
// Shared helpers for TypeScript agent fixture scripts (AAASM-1514 / F116 ST-B).
//
// Each fixture script imports this module for standardised env-var reading and
// structured JSON-line output. The Rust E2E harness (e2e_sdk_node.rs) and the
// developer runner (run_agents_ts.sh) both invoke fixture scripts via:
//
//   AA_GATEWAY_ADDR=<host:port> AA_AGENT_ID=<id> pnpm exec tsx <script>
//
// Selftest mode (AA_SELFTEST=1) emits synthetic events without connecting to
// a real gateway, so scripts can be exercised hermetically in CI.

import { randomBytes } from "node:crypto";

export interface AgentConfig {
  gatewayAddr: string;
  agentId: string;
  task: string;
  proxyAddr?: string; // undefined = Layer 2 inactive
}

export function loadConfig(): AgentConfig {
  const gatewayAddr = process.env.AA_GATEWAY_ADDR;
  if (!gatewayAddr) {
    console.error("error: AA_GATEWAY_ADDR required");
    process.exit(2);
  }
  return {
    gatewayAddr,
    agentId: process.env.AA_AGENT_ID ?? `e2e-${randomBytes(4).toString("hex")}`,
    task: process.env.AA_TASK ?? "noop",
    proxyAddr: process.env.AA_PROXY_ADDR,
  };
}

export function emit(event: Record<string, unknown>): void {
  console.log(JSON.stringify(event));
}

File: aa-integration-tests/tests/fixtures/agents/python/_shared.py
"""Shared env-var loading and event emission helpers for agent fixture scripts."""

from __future__ import annotations

import json
import os
import sys
import uuid


def load_config() -> dict:
    """Load runtime config from environment variables.

    In selftest mode (AA_SELFTEST=1) the gateway address is not required.
    In real mode AA_GATEWAY_ADDR must be set.
    """
    is_selftest = os.environ.get("AA_SELFTEST") == "1"
    gateway = os.environ.get("AA_GATEWAY_ADDR")
    if not gateway and not is_selftest:
        print("error: AA_GATEWAY_ADDR required", file=sys.stderr)
        sys.exit(2)
    return {
        "gateway_addr": gateway or "",
        "agent_id": os.environ.get("AA_AGENT_ID", f"e2e-{uuid.uuid4().hex[:8]}"),
        "task": os.environ.get("AA_TASK", "noop"),
        "proxy_addr": os.environ.get("AA_PROXY_ADDR"),
    }


def emit(event: dict) -> None:
    """Print a JSON event line to stdout."""
    print(json.dumps(event), flush=True)
